tests/test_api.py

```diff
@@ -3046,51 +3047,53 @@ def mock_get_image_stream(*args, **kwargs):
         build_response = osbs_obj._create_build_config_and_build(build_request)
         assert build_response.json == {'spam': 'maps'}
 
-    @pytest.mark.parametrize(('release_value', 'exc'), [
-        ('1release with space', OsbsValidationException),
-        ('1release_with/slash', OsbsValidationException),
-        ('1release_with-dash', OsbsValidationException),
-        ('release.1', OsbsValidationException),
-        ('1release.1.', OsbsValidationException),
-        ('1release.1_', OsbsValidationException),
-        ('1release..1', OsbsValidationException),
-        ('1release__1', OsbsValidationException),
-        ('1.release_with5', None),
-        ('1_release_with5', None),
-        ('123.54.release_with5', None),
-        ('123.54.release_withalpha', None),
-        ('', None),
+    @pytest.mark.parametrize(('label_type', 'label_value', 'raise_exception'), [
+        (Labels.LABEL_TYPE_RELEASE, '1release with space', OsbsValidationException),
+        (Labels.LABEL_TYPE_RELEASE, '1release_with/slash', OsbsValidationException),
+        (Labels.LABEL_TYPE_RELEASE, '1release_with-dash', OsbsValidationException),
+        (Labels.LABEL_TYPE_RELEASE, 'release.1', OsbsValidationException),
+        (Labels.LABEL_TYPE_RELEASE, '1release.1.', OsbsValidationException),
+        (Labels.LABEL_TYPE_RELEASE, '1release.1_', OsbsValidationException),
+        (Labels.LABEL_TYPE_RELEASE, '1release..1', OsbsValidationException),
+        (Labels.LABEL_TYPE_RELEASE, '1release__1', OsbsValidationException),
+        (Labels.LABEL_TYPE_RELEASE, '1.release_with5', None),
+        (Labels.LABEL_TYPE_RELEASE, '1_release_with5', None),
+        (Labels.LABEL_TYPE_RELEASE, '123.54.release_with5', None),
+        (Labels.LABEL_TYPE_RELEASE, '123.54.release_withalpha', None),
+        (Labels.LABEL_TYPE_RELEASE, '', None),
+        (Labels.LABEL_TYPE_VERSION, '1versionwith space', None),
+        (Labels.LABEL_TYPE_VERSION, '1version_with/slash', None),
+        (Labels.LABEL_TYPE_VERSION, '1version_with-dash', OsbsValidationException),
+        (Labels.LABEL_TYPE_VERSION, 'version.1', None),
+        (Labels.LABEL_TYPE_VERSION, '1version.1.', None),
+        (Labels.LABEL_TYPE_VERSION, '', None),
     ])
-    def test_release_label_validation(self, release_value, exc):
-        with NamedTemporaryFile(mode='wt') as fp:
-            fp.write(dedent("""\
-                [general]
-                build_json_dir = inputs
-                [default]
-                build_from = image:buildroot:latest
-                openshift_url = /
-                """))
-            fp.flush()
-            config = Configuration(fp.name)
-            osbs_obj = OSBS(config, config)
+    def test_release_and_version_label_validation(self, osbs, label_type, label_value,
```

Comment:
This test case tests invalid release and version label only, but the name `test ... validation` is too generic. Suggest to rename it to something specifically like `test_raise_error_if_release_or_version_label_is_invalid`.

Reply:
sure